### PR TITLE
fix: MTL pair and triplet training with and without gradient accumulation

### PR DESCRIPTION
**Pull request by gmastrapas**

### CODE REVIEW

File: src/training/embloss.py [hunk 1]
```diff
@@ -362,6 +353,78 @@ def input_type(self):
...
+            loss /= batch_count
+            if self._single_info_nce:
+                loss += self._alpha * self._info_nce_loss(
+                    torch.cat(anchor_embeddings),
+                    torch.cat(pos_embeddings),
+                    torch.cat(neg_embeddings),
+                )
+            weighted_loss += loss * weight
+
+        return weighted_loss
```
Comment (bwanglzu) [inline src/training/embloss.py:425]: if you want to use the MRLMultiCELoss, also use MRLInfoNCELoss, otherwise just use the normal MultiCELoss

**Review by bwanglzu — COMMENTED** [review]
in general looks good to me, I'm not sure about grad_accum, have you tested with , for example, B=16, GA=2 and B=32, GA=1 and make sure the results are identical?

File: src/training/embloss.py (hunk 1 above)
Comment (gmastrapas) [reply src/training/embloss.py:425]: I dont intend to, I just copied the latest file from the large scale training repo

**Comment by gmastrapas** [comment]
> in general looks good to me, I'm not sure about grad_accum, have you tested with , for example, B=16, GA=2 and B=32, GA=1 and make sure the results are identical?

what should I check for?

**Review by koukandre — APPROVED** [review]
